src/components/lyne-toast/lyne-toast.tsx

```diff
@@ -0,0 +1,330 @@
+   */
+  @Event({
+    eventName: 'lyne-toast_did-dismiss'
+  }) public didDismiss: EventEmitter<InterfaceOverlayEventDetail>;
+
+  /**
+   * Emitted after the toast has presented.
+   */
+  @Event({
+    eventName: 'lyne-toast_did-present'
+  }) public didPresent: EventEmitter<void>;
+
+  /**
+   * Emitted before the toast has dismissed.
+   */
+  @Event({
+    eventName: 'lyne-toast_will-dismiss'
+  }) public willDismiss: EventEmitter<InterfaceOverlayEventDetail>;
+
+  /**
+   * Emitted before the toast has presented.
+   */
+  @Event({
+    eventName: 'lyne-toast_will-present'
+  }) public willPresent: EventEmitter<void>;
+
+  /**
+   * Internal timeout.
+   */
+  private _durationTimeout: NodeJS.Timeout;
+
+  /**
+   * Internal toast configuration;
+   * value is merged between the default and the public one.
+   */
+  private _internalConfig: InterfaceToastConfiguration;
+
+  /**
+   * Used to check if the icon slot needs to be rendered.
+   */
+  private _hasIconSlot: boolean;
+
+  /**
+   * Setup the configuration and prepare the overlay.
+   */
+  public connectedCallback(): void {
+    const defaultConfig: InterfaceToastConfiguration = {
+      ariaLivePoliteness: 'polite',
+      horizontalPosition: 'center',
+      message: null,
+      timeout: 6000,
+      verticalPosition: 'bottom'
+    };
+
+    this._internalConfig = {
+      ...defaultConfig,
+      ...this.config
+    };
+    prepareOverlay(this.el);
+  }
+
+  /**
+   * Evaluate if the icon slot is provided by consumers.
+   */
+  public componentWillLoad(): void {
+    this._hasIconSlot = Boolean(this.el.querySelector('[slot="icon"]'));
+  }
+
+  /**
+   * Present the toast overlay after it has been created.
+   */
+  @Method()
+  public async present(): Promise<void> {
+    await present(this, toastEnterAnimation);
+    if (this._internalConfig.timeout > 0) {
+      this._durationTimeout = setTimeout(() => this.dismiss(undefined, 'timeout'), this._internalConfig.timeout);
+    }
+  }
+
+  /**
+   * Dismiss the toast overlay after it has been presented.
+   *
+   * @param data Any data to emit in the dismiss events.
+   * @param role The role of the element that is dismissing the toast.
+   * Example:
+   * `"cancel"` for close icon
+   * `"timeout"` for auto-close
+   */
+  @Method()
+  public dismiss(data?: any, role?: string): Promise<boolean> {
+    if (this._durationTimeout) {
+      clearTimeout(this._durationTimeout);
+    }
+
+    return dismiss(this, data, role, toastLeaveAnimation);
+  }
+
+  /**
+   * Returns a promise that resolves when the toast did dismiss.
+   */
+  @Method()
+  public onDidDismiss<T = any>(): Promise<InterfaceOverlayEventDetail<T>> {
+    return eventMethod(this.el, 'lyne-toast_did-dismiss');
+  }
+
+  /**
+   * Returns a promise that resolves when the toast will dismiss.
+   */
+  @Method()
+  public onWillDismiss<T = any>(): Promise<InterfaceOverlayEventDetail<T>> {
+    return eventMethod(this.el, 'lyne-toast_will-dismiss');
+  }
+
```

Comment:
Hm I don't see the reason why those methods are used? Is it only because of the interface? what are de differences to didDismiss and WillDismiss?

Reply:
They came from Ionic: from what I could see, onDidDismiss is mandatory because it's used in the focusPreviousElementOnDismiss method in overlay.tsx (funny thing: in Ionic it's not defined in OverlayInterface, so if you forget it in an overlay component, the focusPreviousElementOnDismiss method breaks); onWillDismiss appears not to be used explicitly.

For both, I assume they are defined because they might be useful in framework implementations, such as Angular, instead of attaching a listener to the event, something like:
`<my-overlay-comp [config] = 'config' (onWillDismiss)="onWillDismissHandler($event)"> </my-overlay-comp>`
I took a look at the Ionic codebase but I'm not 100% sure about it.